week_11/진명재/1368.java

```diff
@@ -0,0 +1,102 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.StringTokenizer;
+import java.util.stream.IntStream;
+
+public class 1368 {
+    static int[] g;
+    static int[] arr;
+
+    public static void main(String[] args) throws Exception {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
+        int N = Integer.parseInt(br.readLine());
+
+        arr = new int[N + 1];
+
+
+        for (int i = 1; i < N + 1; i++) {
+            arr[i] = Integer.parseInt(br.readLine());
+        }
+
+        PriorityQueue<non> que = new PriorityQueue<>(Comparator.comparingInt(non::getV));
+
+        for (int i = 1; i < N + 1; i++) {
+            StringTokenizer st = new StringTokenizer(br.readLine());
+            for (int j = 1; j < N + 1; j++) {
+                int v = Integer.parseInt(st.nextToken());
+                if (i >= j) continue;
+                que.add(new non(i, j, v));
+            }
+        }
+
+        System.out.println(solution(N, que, arr));
```

Comment:
static으로 필드에 저장했으면 파라미터로 들어가면 안됨 (비효율)
```suggestion
        System.out.println(solution(N, que));
```